### ISSUE TITLE
Inconsistent input for score calculation in translation_moe

### ISSUE THREAD

**Issue by jaimeenahn** [pytorch/fairseq#2277]
## 🐛 Bug

In translation_moe, the script calculating the score between results with BPE and without BPE.

### To Reproduce

Steps to reproduce the behavior (**always include the command you ran**):

First download a tokenized version of the WMT'14 En-De test set with multiple references:
```bash
wget dl.fbaipublicfiles.com/fairseq/data/wmt14-en-de.extra_refs.tok
```

Next apply BPE on the fly and run generation for each expert:
```bash
BPE_CODE=examples/translation/wmt17_en_de/code
for EXPERT in $(seq 0 2); do \
    cat wmt14-en-de.extra_refs.tok \
    | grep ^S | cut -f 2 \
    | fairseq-interactive data-bin/wmt17_en_de \
        --path checkpoints/checkpoint_best.pt \
        --beam 1 \
        --bpe subword_nmt --bpe-codes $BPE_CODE \
        --buffer-size 500 --max-tokens 6000 \
        --task translation_moe --user-dir examples/translation_moe/src \
        --method hMoElp --mean-pool-gating-network \
        --num-experts 3 \
        --gen-expert $EXPERT ; \
done > wmt14-en-de.extra_refs.tok.gen.3experts
```

Finally use `score_moe.py` to compute pairwise BLUE and average oracle BLEU:
```bash
python examples/translation_moe/score.py --sys wmt14-en-de.extra_refs.tok.gen.3experts --ref wmt14-en-de.extra_refs.tok
# pairwise BLEU: 48.26
# #refs covered: 2.11
# multi-reference BLEU (leave-one-out): 59.46
```

### Expected behavior

Ideally if we generate the result using BPE, the calculating function should compare the reference which BPE is applied. However, BPE isn't applied to **wmt14-en-de.extra_refs.tok** .

### Environment

 - fairseq Version (e.g., 1.0 or master): Master
 - PyTorch Version (e.g., 1.0) 1.4.0
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): pip install --editable ./
 - Python version: 3.7.7
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: V 100
 - Any other relevant information:

### Additional context

```bash
BPE_CODE=examples/translation/wmt17_en_de/code
for EXPERT in $(seq 0 2); do \
    cat wmt14-en-de.extra_refs.tok \
    | grep ^S | cut -f 2 \
    | fairseq-interactive data-bin/wmt17_en_de \
        --path checkpoints/checkpoint_best.pt \
        --beam 1 \
        --bpe subword_nmt --bpe-codes $BPE_CODE \
        --buffer-size 500 --max-tokens 6000 \
        --task translation_moe --user-dir examples/translation_moe/src \
        --method hMoElp --mean-pool-gating-network \
        --num-experts 3 \
        **--remove-bpe**
        --gen-expert $EXPERT ; \
done > wmt14-en-de.extra_refs.tok.gen.3experts
```

`--remove-bpe` should be added end of generating the script for the reported score on paper.

This can be answer for #2212 as well.

**Comment by lematt1991** [comment]
I'm not sure I understand, you want to compute BLUE *with* BPE?  If so, can't you just apply BPE to your reference and call the same script?

**Comment by myleott** [comment]
If I understand right, @jaimeenahn is pointing out that we should have `--remove-bpe` in the instructions here: https://github.com/pytorch/fairseq/tree/master/examples/translation_moe#evaluate. I'll add that

**Comment by jaimeenahn** [comment]
@myleott That's correct.

Without `--remove-bpe`, the average multi-reference BLEU of hMoEup is 35.89, however, reported score in the paper is 62.6. If `--remove-bpe` is applied, the average multi-reference BLEU is magically increased to 60.72 which is acceptable.

**Comment by myleott** [comment]
Fixed in 98b79a42c7f6ec66e6fc3881f70b23087fd50d02

**Comment by zhmingzhaung** [comment]
```
for EXPERT in $(seq 0 2); do \
    cat wmt14-en-de.extra_refs.tok \
    | grep ^S | cut -f 2 \
    | fairseq-interactive $DATA_DIR/ --path $MODELDIR \
        --beam $BEAM_SIZE \
        --bpe subword_nmt --bpe-codes $BPE_CODE \
        --source-lang $SRC \
        --target-lang $TGT  \
        --task translation_moe --user-dir examples/translation_moe/translation_moe_src \
        --method hMoElp --mean-pool-gating-network \
        --num-experts 3 \
        --batch-size $DECODER_BS \
        --buffer-size $DECODER_BS --max-tokens 6000 \
        --remove-bpe \
        --gen-expert $EXPERT ; \
done > wmt14-en-de.extra_refs.tok.gen.3experts


python examples/translation_moe/score.py \
    --sys wmt14-en-de.extra_refs.tok.gen.3experts \
    --ref wmt14-en-de.extra_refs.tok
```

This is the command I used, as u see I add `--remove-bpe` but the output log still is:

> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> pairwise BLEU: 48.35
> #refs covered: 2.63
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> That's 100 lines that end in a tokenized period ('.')
> It looks like you forgot to detokenize your test data, which may hurt your score.
> If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
> average multi-reference BLEU (leave-one-out): 37.67

I don't know what should I do, please help me.
